postgresql/config/functions.sh

```diff
@@ -0,0 +1,106 @@
+init_pgpass() {
+  export PGPASSFILE="{{pkg.svc_config_path}}/.pgpass"
+  chmod 0600 {{pkg.svc_config_path}}/.pgpass
+}
+
+ensure_dir_ownership() {
+  echo 'Making sure hab user owns var, config and data paths'
+  chown -RL hab:hab {{pkg.svc_var_path}}
+  chown -RL hab:hab {{pkg.svc_config_path}}
+  chown -RL hab:hab {{pkg.svc_data_path}}
+}
+
+write_local_conf() {
+  echo 'Writing postgresql.local.conf file based on memory settings'
+  cat > {{pkg.svc_config_path}}/postgresql.local.conf<<LOCAL
+# Auto-generated memory defaults created at service start by Habitat
+effective_cache_size=$(awk '/MemTotal/ {printf( "%.0f\n", ($2 / 1024 / 4) *3 )}' /proc/meminfo)MB
+shared_buffers=$(awk '/MemTotal/ {printf( "%.0f\n", $2 / 1024 / 4 )}' /proc/meminfo)MB
+maintenance_work_mem=$(awk '/MemTotal/ {printf( "%.0f\n", $2 / 1024 / 16 )}' /proc/meminfo)MB
+work_mem=$(awk '/MemTotal/ {printf( "%.0f\n", (($2 / 1024 / 4) *3) / ({{cfg.max_connections}} *3) )}' /proc/meminfo)MB
+temp_buffers=$(awk '/MemTotal/ {printf( "%.0f\n", (($2 / 1024 / 4) *3) / ({{cfg.max_connections}}*3) )}' /proc/meminfo)MB
+LOCAL
+}
+
+write_env_var() {
+  echo "$1" > "{{pkg.svc_config_path}}/env/$2"
+}
+
+write_wale_env() {
+  echo 'Writting environment variables required by wal-e'
+  mkdir -p "{{pkg.svc_config_path}}/env"
+  {{#with cfg.wal-e.aws }}
```

Comment:
When using `#with` it's almost always clearer to add `as |aws|}}`...`{{aws.bucket}}`.